src/modules/core/components/dialog-bottom/dialog-bottom.tsx

```diff
@@ -0,0 +1,159 @@
+// Heads up! This component designed to use only on mobile screens.
+
+import { type FC, useCallback, useEffect, useRef } from 'react';
+import * as RadixDialog from '@radix-ui/react-dialog';
+import {
+  Controller,
+  animated,
+  easings,
+  type SpringConfig,
+} from '@react-spring/web';
+import { useBoolean, useScreen } from 'usehooks-ts';
+// hooks
+import { useDeviceType } from '@/modules/core/hooks/use-device-type';
+import { useTouchDirectionDrag } from '@/modules/core/hooks/use-touch-direction-drag';
+// utils
+import { reassignCssProperties } from '@/modules/core/utils/css.utils';
+
+import type { BottomActionsProps } from './dialog-bottom.interface';
+import styles from './dialog-bottom.module.scss';
+
+const DRAG_OFFSET_CLOSE_BREAKDOWN = 100;
+const IN_OUT_ANIMATION_DURATION = 200;
+const OVERLAY_OPACITY = 0.6;
+
+const animationConfig: SpringConfig = {
+  duration: IN_OUT_ANIMATION_DURATION,
+  easing: easings.easeInOutCubic,
+};
+
+const contentAnimationController = new Controller({
+  bottom: '0px',
+  config: animationConfig,
+});
+const overlayAnimationController = new Controller({
+  opacity: 0,
+  config: animationConfig,
+});
+
+export const DialogBottom: FC<BottomActionsProps> = ({
+  isOpen,
+  onOpenChange,
+  children,
+}) => {
+  const deviceType = useDeviceType();
+  // state
+  const _isOpen = useBoolean(isOpen); // internal open state to keep modal visible until animation end
+  const screen = useScreen();
+  // refs
+  const contentRef = useRef<HTMLDivElement>(null);
+  const overlayRef = useRef<HTMLDivElement>(null);
+
+  // update _isOpen immediately to true
+  useEffect(() => {
+    if (isOpen) {
+      _isOpen.setTrue();
+    }
+  }, [isOpen, _isOpen.setTrue]);
+
+  // in/out animation control
+  useEffect(() => {
+    if (!screen) return;
+
+    // start 'open animation'
+    if (isOpen && _isOpen.value) {
+      contentAnimationController.set({ bottom: `-${screen.height}px` });
+      overlayAnimationController.set({ opacity: 0 });
+
+      Promise.all([
+        contentAnimationController.start({ bottom: '0px' }),
+        overlayAnimationController.start({ opacity: OVERLAY_OPACITY }),
+      ]);
+    }
+
+    // start 'close animation'
+    if (!isOpen && _isOpen.value) {
+      Promise.all([
+        contentAnimationController.start({ bottom: `-${screen.height}` }),
+        overlayAnimationController.start({ opacity: 0 }),
+      ]).finally(() => {
+        _isOpen.setFalse();
+      });
+    }
+  }, [isOpen, _isOpen, screen]);
```

Comment:
Looks interesting, I assume that's all working well as I saw it on the app, so I think we can leave it like that, you can resolve this comment right away.